salt/returners/__init__.py

```diff
@@ -137,7 +137,7 @@ def _fetch_option(cfg, ret_config, virtualname, attr_name):
     if not ret_config:
         # Using the default configuration key
         if isinstance(cfg, dict):
-            return c_cfg.get(attr_name, cfg.get(default_cfg_key))
+            return c_cfg.get(default_cfg_key, cfg.get(attr_name))
```

Comment:
Here is what I understand - Because of line 131 check if execution flow goes in 131 and then comes to line 140, then both c_cfg and cfg are going to be same. In this case it's better to check for a more specific key (e.g. returner.postgres.user) before checking for a generic key (e.g. user).

I've tested with a scheduler as well, in this case I've grains been updated every 15 minutes and that job return data is saved correctly even after this change. Please note that saving of return data from a job created by a scheduler caused event was NOT an issue before either.